### PR TITLE
fix crash + improve events

### PR DESCRIPTION
**Pull request by yannsucc**
- fix crash on tcp client
- improve events client/server

### CODE REVIEW

File: mphx/client/EventManager.hx [hunk 1]
```diff
@@ -11,17 +11,44 @@ class EventManager
...
+	
+	/**
+	 * Remove an event if exist
+	 * @param	eventName
+	 */
+	public function remove(eventName:String)
+	{
+		if (eventMap.exists(eventName)) 
+		{
+			eventMap.set(eventName,null);
```
Comment (anissen) [inline mphx/client/EventManager.hx:33]: This is redundant, you remove the entry in the next line

File: mphx/client/EventManager.hx [hunk 2]
```diff
@@ -11,17 +11,44 @@ class EventManager
...
+	 */
 	public function callEvent(eventName:String, data:Dynamic)
 	{
 		//If an event with that name exists.
-		if (eventMap.exists(eventName)) {
+		if (eventMap.exists(eventName)) 
+		{
 			//See if the event should be called with or without the sender.
-			eventMap.get(eventName)(data);
+			if(eventMap.get(eventName) != null)
```
Comment (anissen) [inline mphx/client/EventManager.hx:50]: This check should not be here. If anywhere, it should be in the `on`-function throwing an error if `event` is null.

**Comment by 5Mixer** [comment]
Thanks. I've been busy recently so I won't be able to test, but in 2 days I have 2 weeks to fix this project and make sure it's not broken. Thanks again.

File: mphx/client/EventManager.hx (hunk 1 above)
Comment (yannsucc) [inline mphx/client/EventManager.hx:33]: forget to remove this line.

File: mphx/client/EventManager.hx (hunk 2 above)
Comment (yannsucc) [inline mphx/client/EventManager.hx:50]: Hum, depends if you allow a null callback... Might be usefull for desactive temporary an event without destroy it. 
About throwing an error, i'm not sure that it's necessary. In my mind, it's not an important error. Maybe return true/false if the add had work ? I prefer let @5Mixer decide about this point.

**Comment by 5Mixer** [comment]
For starters, events will probably be improved alot soon. Regex/custom events is certainly on my mental todo.

In terms of what to do if calling an event that doesn't exist, I think it should just fail gracefully and return false. Events are really only called by internal library code when data is recieved. That data has a type, and tries to run an event with that type here. Throwing an error is bad because the server should be able to send events, and not rely on clients to listen to them. Perhaps it could return false if it doesn't exist, and then, `#if debug` then output something to console ('Recieved an event but no listener was registered'). That could perhaps be useful for typos (server sends event with type `PlayerMove` and client is listening for `PlayerUpdate`)

I'll get to this soon, but it's not too important yet.

**Pull request merged**